File: ex1.py
#explicações importantes:
#tipos de dados: int, float, str e bool
#int = numeros inteiros
#float = numeros com ponto flutuante
#str = strings, textos
#bool = booleano, true ou false

#################################################################################################################
#EXERCICIOS DE PYTHON
#################################################################################################################

#calculadora basica

num1 = 10
num2 = 20

soma = num1 + num2
subtracao = num1 - num2
multiplicacao = num1 * num2
divisao = num1 / num2
resto_da_divisao = num1 % num2

print(f"Soma = ", soma)
print(f"Subtração = ", subtracao)
print(f"Multiplicação = ", multiplicacao)
print(f"Divisão = ", divisao)
print(f"Resto da divisão = ", resto_da_divisao)

#exercicio mensagem personaliazada

nome = "Pedro"
idade = 20
altura = 1.75
peso = 75

print(f"Olá, meu nome é {nome}, tenho {idade} anos, minha altura é {altura}m e meu peso é {peso}kg")

#exercicio calcular maioridade 

idade = 20

if idade < 18:
    print("Menor de idade")

elif idade == 18:
    print("Possui 18 anos")

else:
   print("Maior de idade")

#exercicio calcular imc

peso = 76
altura = 1.75

imc = peso / (altura ** 2)

if imc < 18.5:
    print("Abaixo do peso")

elif imc >= 18.5 and imc < 24.9:
    print("Peso normal")

elif imc <= 25 and imc < 29.9:
    print("Sobrepeso")

elif imc <= 30 and imc < 34.9:
    print("Obesidade grau 1")

elif imc <= 35 and imc < 39.9:
    print("Obesidade grau 2")

elif imc >= 40:
    print("Obesidade grau 3")

elif imc > 60:
    print("Obesidade mórbida")

else:
    print("Morreu")

print(f"Seu imc: {imc:2f}")

#exercicio calcular par ou impar

num = 10

if num % 2 == 0:
    print("Numero par")

else:
    print("Numero impar")

#exercicio conversor de metros

metros = 100

centimetros = metros * 100
milimetros = metros * 1000
km = metros / 1000

print(f"{metros} metros equivalem a {centimetros} centimetros, {milimetros} milimetros e {km} km")

#exercicio converter celsius para fahrenheit
celcius = 0

fahrenheit = celcius * 1.8 + 32

print(f"a temperatura em celsius é {celcius} e em fahrenheit é {fahrenheit}")

#exercicio media de tres notas (aprovado e reprovado)

nota1 = 9.2
nota2 = 7.6
nota3 = 8.9

media = (nota1 + nota2 + nota3) / 3

if media >= 7:
 print("Aprovado")

elif media >= 5 and media <= 6.9:
 print("Recuperação")

elif media < 5:
 print("Reprovado")

#exercicio desconto em produto

produto = 100
desconto = produto * 20 / 100 #desconto de 20%
valor_final = produto - desconto

print(f"o produto custa {produto} reais, com o desconto aplicado, passa a custar {valor_final} reais.")

##exercicio verificar multiplos

numero1 = 5
numero2 = 10

if numero1 % numero2 == 0 or numero2 % numero1 == 0:
    print("multiplos.")

else:
    print("nao sao multiplos.")

##exercicio calculadora com input e operação

numero1 = int(input("Digite o primeiro numero: "))
numero2 = int(input("Digite o segundo numero: "))

operacao = input("Digite a operação: ")

if operacao == "+":
    print("A soma entre os dois numeros é: ", numero1 + numero2)

elif operacao == "-":
    print("A subtração entre os dois numeros é: ", numero1 - numero2)

elif operacao == "*":
    print("a multiplicação entre os dois numeros é: ", numero1 * numero2)

elif operacao == "/" and numero2 != 0:
    print("a divisão entre os dois numeros é: ", numero1 / numero2)

else:
    print("ERRO")

##exercicio login simples (usuario e senha)

usuario = input("Digite seu user: ")
senha = input("Digite sua senha: ")
usuario_correto = "admin"
senha_correta = "1234"

if usuario != usuario_correto or senha != senha_correta:
    print("ACESSO NEGADO")
    
else:
    print("ACESSO PERMITIDO")

##exercicio calcular idades

idade = 23

if idade > 0 and idade <= 11:
    print("Você é criança.")
    
elif idade >= 12 and idade <= 17:
    print("Você é adolescente.")
    
elif idade >= 18 and idade <= 59:
    print("Você é adulto.")
    
elif idade > 60:
    print("Você é idoso.")

##exercicio numero negativo, positivo ou zero

num = 0

if num > 0:
    print("Numero positivo")
    
elif num < 0:
    print("Numero negativo")
    
elif num == 0:
    print("ZERO")

##exercicio maior de tres numeros 

num1 = 10
num2 = 20
num3 = 30

if num1 > num2 and num1 > num3:
    print("o primeiro numero é maior")
    
elif num2 > num1 and num2 > num3:
    print("o segundo numero é maior")
    
else:
    print("o terceiro numero é maior")

##exercicio verificar ano bissexto

ano = 2024

if ano % 400 == 0:
    print("bissexto")
    
elif ano % 100 == 0:
    print("nao bissexto")
    
elif ano % 4 == 0:
    print("bissexto")
    
else:
    print("nao bissexto")

File: ex1.js
//vou fazer os mesmos exercicios de python em javascript
//--------------------------------------------------------

//exercicio calculadora basica

let num1 = 10;
let num2 = 20;

const soma = num1 + num2;
const subtracao = num1 - num2;
const multiplicacao = num1 * num2;
const divisao = num1 / num2;
const restoDaDivisao = num1 % num2;

console.log(`a soma dos dois numeros é: ${soma}`);
console.log(`a subtraçao dos dois numeros é: ${subtracao}`);
console.log(`a multiplicação dos dois numeros é: ${multiplicacao}`);
console.log(`a divisão dos dois numeros é: ${divisao}`);
console.log(`o resto da divisao entre os dois numeros é: ${restoDaDivisao}`);

//exercicio mensagem personalizada

let nome = "pedro";
let idade = 20;
let altura = 1.75;
let peso = 75;

console.log(`Olá, meu nome é ${nome}, eu tenho ${idade} anos, minha altura é ${altura} e eu peso ${peso}kg.`);

//exercicio calcular maioridade

let idade1 = 20;

if (idade < 18) {
    console.log("Menor de idade.")
} 
else if (idade === 18) {
    console.log("Possui 18 anos.")
} 
else {
    console.log("Maior de idade.")
}

//exercicio calcular imc

let pesoIMC = 76;
let alturaIMC = 1.75;

imc = peso / (altura ** 2);

if (imc < 18.5) {
    console.log("Abaixo do peso");
}
else if (imc >= 18.5 && imc < 24.9) {
    console.log("Peso normal");
}
else if (imc <= 25 && imc < 29.9) {
    console.log("Sobrepeso");
}
else if (imc <= 30 && imc < 34.9) {
    console.log("Obesidade grau 1");
}
else if (imc <= 35 && imc < 39.9) {
    console.log("Obesidade grau 2");
}
else if (imc >= 40) {
    console.log("Obesidade grau 3");
}
else if (imc > 60) {
    console.log("Obesidade mórbida");
}
else {
    console.log("Morreu.");
}

console.log(`Seu IMC é ${imc.toFixed(2)}`);

//exercicio calcular par ou impar 

let num9 = 10;

if (num9 % 2 === 0) {
    console.log("Numero par")
}
else {
    console.log("Numero impar");
}

//exercicio conversor de metros

let metros = 100
let centimetros = metros * 100;
let milimetros = metros * 1000;
let km = metros / 1000;

console.log(`${metros} metros equivalem a ${centimetros} centimetros, ${milimetros} milimetros e ${km}km.`);

//exercicio converter celsius para fahrenheit
//formula: fahrenheit = celcius * 1.8 + 32

let celsius = 20;
fahrenheit = celsius * 1.8 + 32;

console.log(`a temperatura em celsius é ${celsius}, e em fahrenheit é ${fahrenheit}.`);

//exercicio media de tres notas (aprovado e reprovado)

const nota1 = 8;
const nota2 = 10;
const nota3 = 7;

media = (nota1 + nota2 + nota3) / 3;

if (media >= 7) {
    console.log("Aprovado.");
}
else if (media >= 5 && media < 6.9) {
    console.log("Recuperação");
}
else if (media < 5) {
    console.log("Reprovado.");
}

//exercicio desconto em produtos

let produto = 100;
desconto = produto * 20 / 100;
valorFinal = produto - desconto;

console.log(`o produto originalmente custa ${produto} reais, com o desconto de ${desconto} reais, ele passa a custar ${valorFinal} reais.`);

//exercicio verificar multiplos

let numero1 = 5;
let numero2 = 10;

if (numero1 % numero2 === 0 || numero2 % numero1 === 0) {
    console.log("numeros multiplos")
}
else 
    console.log("nao sao multiplos")